src/assisted_test_infra/test_infra/helper_classes/cluster.py

```diff
@@ -801,10 +803,46 @@ def download_image(self, iso_download_path: str = None) -> Path:
             )
         return self._infra_env.download_image(iso_download_path)
 
+    # Downloads the iPXE script from assisted service to a local file
+    def prepare_local_ipxe_server(self, infra_env_id):
+        assisted_service_url = utils.get_assisted_service_url_by_configmap("assisted-service-config", "assisted-installer")
+        remote_ipxe_path = f"/api/assisted-install/v2/infra-envs/{infra_env_id}/downloads/files?file_name=ipxe-script"
+        ipxe_download_url = f"{assisted_service_url}{remote_ipxe_path}"
+
+        local_ipxe_script_file = f"{self._config.local_ipxe_server_artifacts_path}/{self._config.cluster_name}"
+
+        if not os.path.exists(local_ipxe_script_file):
+            log.info("Local iPXE server path doesn't exist, creating the folder: %s", local_ipxe_script_file)
+            utils.recreate_folder(os.path.dirname(local_ipxe_script_file), force_recreate=False)
+        utils.download_file(ipxe_download_url, local_ipxe_script_file, False)
+
+    # Runs a local python server that hosts the iPXE scripts 
+    def run_local_ipxe_server(self):
+        ip = re.search('\d+\.\d+\.\d+\.\d+',self._config.local_ipxe_server_url).group(0)
+        log.info("the ip %s", ip)
+        run_ipxe_server_cmd = f'''python3 {self._config.local_ipxe_server_script_path} \
+            --dir {self._config.local_ipxe_server_artifacts_path} \
+            --ip {ip} \
+            --port {self._config.local_ipxe_server_port}'''
+        log.info("the cmd %s", run_ipxe_server_cmd)
+        utils.run_command(run_ipxe_server_cmd, run_in_background=True)
+
```

Comment:
Also here, create an ipxe controller as I mentioned before, and put those methods there.

Reply:
Created the controller!